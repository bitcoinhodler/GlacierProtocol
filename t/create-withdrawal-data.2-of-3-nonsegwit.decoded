{
  "txid": "47454477df02e9d8a5cf2c2d01063af7c7b97c57c30e7991c4cda52d603d31c3",
  "hash": "47454477df02e9d8a5cf2c2d01063af7c7b97c57c30e7991c4cda52d603d31c3",
  "version": 2,
  "size": 337,
  "vsize": 337,
  "weight": 1348,
  "locktime": 0,
  "vin": [
    {
      "txid": "f5cf28849c1207e535329543b7264d65dc976b001ed564d4032243ae53670da5",
      "vout": 0,
      "scriptSig": {
        "asm": "0 3044022018e5fe19eedf2abec9b4a74a8ab0e973625af14eeed940df519f4db2653a8d55022052dcd605e118c841307f8575b8813f2a8b415276252fa03dc70e961cdcadf7d8[ALL] 3044022039a14fc5049dc00370bf635e0f6f74f79d0dc498f65c0e58027fa3b80728bd4402201b0fbaff27bbcc59bebec588505319e3830a542f9847cd51d041fb9edc3a58a6[ALL] 52210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d653ae",
        "hex": "00473044022018e5fe19eedf2abec9b4a74a8ab0e973625af14eeed940df519f4db2653a8d55022052dcd605e118c841307f8575b8813f2a8b415276252fa03dc70e961cdcadf7d801473044022039a14fc5049dc00370bf635e0f6f74f79d0dc498f65c0e58027fa3b80728bd4402201b0fbaff27bbcc59bebec588505319e3830a542f9847cd51d041fb9edc3a58a6014c6952210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d653ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.19996630,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 b6c7733e23a271e57ca16e5cd885174917968db9 OP_EQUALVERIFY OP_CHECKSIG",
        "desc": "addr(mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99)#fxf0ktmw",
        "hex": "76a914b6c7733e23a271e57ca16e5cd885174917968db988ac",
        "address": "mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99",
        "type": "pubkeyhash"
      }
    }
  ]
}
