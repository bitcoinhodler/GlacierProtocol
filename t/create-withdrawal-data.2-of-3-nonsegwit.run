#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2NGPJX8kzdRpAQJuZWMpCBth1umjQFHeFcz
52210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d653ae
mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
1
02000000018995521ee8ae02f505cf47410732398af0ac2486c5ce17d1a906fdc3588edf15010000006a47304402203f027d065045bbb3a50efc3db8d4fd8874556df8dc6c22aa9cf634aaaf40b842022076b2902907d2eda235e2b789831f76dd04b166bc56487007cd3fd34f23ddede60121023cf821f44685d00d22ef099df090c8889bfb1d2a9b295561012e1af74997dd88ffffffff05002d31010000000017a914fdd200f6e02076173292642fd352dc45f849070e8790409700000000001976a914c9c141605f5c5dc920a6d9b283b6f8dc0cedd3ed88ac002d31010000000017a9145b0711b907a5fed5894f083b212a25ae517626a187002d31010000000017a914670badde8a4e4e1f9289e4ea6cf9a93acbe8266387002d31010000000017a91412143b7614ba4cc3da6779930c39fe167bef54e48700000000
2
cVV1MBAgdAnrX8bjQWSJcgZAbVtprR8kgdfez858ejvXs3qZWqn2
cQy39MvkMKeaEvG1wXejdtEvAdU381u9EBBmcKUJWbfn6QL82YvR
10
y

y
INPUT
