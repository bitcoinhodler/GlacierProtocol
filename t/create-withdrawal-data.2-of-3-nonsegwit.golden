Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.20000000 BTC

How many private keys will you be signing this transaction with (at least 2 required)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00003690 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00003690, you have 0.19996310 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.19996630 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.20000000 BTC in unspent inputs from cold storage address 2NGPJX8kzdRpAQJuZWMpCBth1umjQFHeFcz
0.19996630 BTC going to destination address mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
Fee amount: 0.00003370

Signing with private keys: 
cVV1MBAgdAnrX8bjQWSJcgZAbVtprR8kgdfez858ejvXs3qZWqn2
cQy39MvkMKeaEvG1wXejdtEvAdU381u9EBBmcKUJWbfn6QL82YvR


Confirm? (y/n): 
Calculating transaction...


Raw signed transaction (hex):
02000000017870ad718fb9a1dbb84a62168e729f2913fa57eea0234d519e6dba906b94515400000000fc00473044022046b406774b487b8b9756e2a0dbdaecbd48a2ab3e0458b53dfe97b9f9a40c1c650220622aa13f83f2a59df4c267628e4a91b35f4785624536897bb3e833dc6c695e640147304402201977651e7ab908e9c8cfdadae24bb86f1dc1df44d6e8f11240155e8248e8ae4102204447e64c6cb5d293539be31bc4b7531ab194511ac1ab729e9f72441fdc94784b014c6952210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d653aeffffffff01d61f3101000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac00000000

Transaction fingerprint (md5):
02dc3a90577f1d32290902ce59e36585
QR code for transaction written to transaction.png
